src/core/federation.ts

```diff
@@ -1133,14 +1133,16 @@ export class FederatedApp extends App {
         for (let sendsToFedId of config.sendsTo) {
             this.addDownstreamFederate(sendsToFedId);
         }
+        let upstreamConnectionDelaysIndex = 0;
         for (let dependsOnFedId of config.dependsOn) {
             let minOutputConnectionDelay = TimeValue.FOREVER();
-            for (let candidate of config.upstreamConnectionDelays[dependsOnFedId]) {
+            for (let candidate of config.upstreamConnectionDelays[upstreamConnectionDelaysIndex]) {
```

Comment:
I don't think there's any boundary check for `upstreamConnectionDelaysIndex` in the current code. How do you make sure that `upstreamConnectionDelaysIndex` does not go out of index bound (e.g., it doesn't go over the size of the array)?

Reply:
`upstreamConnectionDelaysIndex` does not exceed the size of the array `config.depensOn`. And the size of `config.dependsOn` and `config.upstreamConnectionDelays` are the same because of [this code](https://github.com/lf-lang/lingua-franca/blob/61f6d678bc2530ce14d2e7697b51ec04d8b5ab52/org.lflang/src/org/lflang/federated/extensions/TSExtension.java#L178).